core/src/main/java/org/elasticsearch/index/indexing/IndexSlowLog.java

```diff
@@ -59,46 +55,40 @@
+        long indexWarnThreshold = settings.getAsTime(INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_WARN, TimeValue.timeValueNanos(IndexSlowLog.this.indexWarnThreshold)).nanos();
+        if (indexWarnThreshold != IndexSlowLog.this.indexWarnThreshold) {
+            IndexSlowLog.this.indexWarnThreshold = indexWarnThreshold;
+        }
+        long indexInfoThreshold = settings.getAsTime(INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_INFO, TimeValue.timeValueNanos(IndexSlowLog.this.indexInfoThreshold)).nanos();
+        if (indexInfoThreshold != IndexSlowLog.this.indexInfoThreshold) {
+            IndexSlowLog.this.indexInfoThreshold = indexInfoThreshold;
+        }
+        long indexDebugThreshold = settings.getAsTime(INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_DEBUG, TimeValue.timeValueNanos(IndexSlowLog.this.indexDebugThreshold)).nanos();
+        if (indexDebugThreshold != IndexSlowLog.this.indexDebugThreshold) {
+            IndexSlowLog.this.indexDebugThreshold = indexDebugThreshold;
+        }
+        long indexTraceThreshold = settings.getAsTime(INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_TRACE, TimeValue.timeValueNanos(IndexSlowLog.this.indexTraceThreshold)).nanos();
+        if (indexTraceThreshold != IndexSlowLog.this.indexTraceThreshold) {
+            IndexSlowLog.this.indexTraceThreshold = indexTraceThreshold;
+        }
 
-            String level = settings.get(INDEX_INDEXING_SLOWLOG_LEVEL, ShardSlowLogIndexingService.this.level);
-            if (!level.equals(ShardSlowLogIndexingService.this.level)) {
-                ShardSlowLogIndexingService.this.indexLogger.setLevel(level.toUpperCase(Locale.ROOT));
-                ShardSlowLogIndexingService.this.deleteLogger.setLevel(level.toUpperCase(Locale.ROOT));
-                ShardSlowLogIndexingService.this.level = level;
-            }
+        String level = settings.get(INDEX_INDEXING_SLOWLOG_LEVEL, IndexSlowLog.this.level);
+        if (!level.equals(IndexSlowLog.this.level)) {
+            IndexSlowLog.this.indexLogger.setLevel(level.toUpperCase(Locale.ROOT));
+            IndexSlowLog.this.deleteLogger.setLevel(level.toUpperCase(Locale.ROOT));
+            IndexSlowLog.this.level = level;
+        }
 
-            boolean reformat = settings.getAsBoolean(INDEX_INDEXING_SLOWLOG_REFORMAT, ShardSlowLogIndexingService.this.reformat);
-            if (reformat != ShardSlowLogIndexingService.this.reformat) {
-                ShardSlowLogIndexingService.this.reformat = reformat;
-            }
+        boolean reformat = settings.getAsBoolean(INDEX_INDEXING_SLOWLOG_REFORMAT, IndexSlowLog.this.reformat);
+        if (reformat != IndexSlowLog.this.reformat) {
+            IndexSlowLog.this.reformat = reformat;
         }
     }
 
-    @Inject
-    public ShardSlowLogIndexingService(ShardId shardId, @IndexSettings Settings indexSettings, IndexSettingsService indexSettingsService) {
-        super(shardId, indexSettings);
-
+    IndexSlowLog(ESLogger logger, Settings indexSettings) {
```

Comment:
I think this changes the logger name we have. Before it was `index.indexing.slowlog.index` and now we will have `index.indexing.index` we should adapt the getLogger calls below.